doc/languages-frameworks/rust.section.md

```diff
@@ -397,9 +397,13 @@ In this scenario, the `checkPhase` will be ran in `debug` mode as well.
 
 Some packages may use custom scripts for building/installing, e.g. with a `Makefile`.
 In these cases, it's recommended to override the `buildPhase`/`installPhase`/`checkPhase`.
-
 Otherwise, some steps may fail because of the modified directory structure of `target/`.
 
+If you want to run hooks from other build systems, instead of overriding the phases,
+you can set `dontCargoBuild`/`dontCargoInstall`/`dontCargoCheck` to `true` to
+use the [default phases from `stdenv`](#sec-stdenv-phases), which execute make
+targets and can be controlled by various variables in the derivation.
```

Comment:
```suggestion
If you have to use build system other than cargo, instead of overriding the phases,
you can set `dontCargoBuild`/`dontCargoInstall`/`dontCargoCheck` to `true` to
use the [default phases from `stdenv`](#sec-stdenv-phases).
```